### PR TITLE
Adds ability to select premium plan on signup

### PR DESCRIPTION
**Pull request by broskoski**
signup

### CODE REVIEW

File: react/components/Billing/components/BillingForm/index.js [hunk 1]
```diff
@@ -322,7 +340,7 @@ class BillingForm extends PureComponent {
                 onClick={this.handleSubmit}
                 disabled={(
                   (!customer.default_credit_card) ||
-                  (operations.length === 0)
+                  (operations.length === 0 && !this.props.plan_id)
```
Comment (dzucconi) [inline react/components/Billing/components/BillingForm/index.js]: I'm not sure I follow the reasoning here? Can you explain?

File: react/components/Billing/components/BillingForm/index.js [hunk 2]
```diff
@@ -39,20 +39,32 @@ class BillingForm extends PureComponent {
...
+    onSuccess: () => null,
   }
 
   state = {
     mode: 'resting',
     errorMessage: null,
     operations: [],
 
-    plan_id: this.props.me.customer.plan.id,
+    plan_id: this.props.plan_id || this.props.me.customer.plan.id,
```
Comment (dzucconi) [inline react/components/Billing/components/BillingForm/index.js]: I don't think this needs to be here if you are calling handlePlan in the `componentDidMount`?

File: react/components/Billing/components/BillingForm/index.js (hunk 1 above)
Comment (broskoski) [reply react/components/Billing/components/BillingForm/index.js]: Taking this out

File: react/components/Billing/components/BillingForm/index.js (hunk 2 above)
Comment (broskoski) [reply react/components/Billing/components/BillingForm/index.js]: And this too

**Pull request merged**